mm2src/coins/z_coin.rs

```diff
@@ -1935,8 +2145,18 @@ impl InitWithdrawCoin for ZCoin {
         let received_by_me = big_decimal_from_sat_unsigned(data.received_by_me, self.decimals());
         let spent_by_me = big_decimal_from_sat_unsigned(data.spent_by_me, self.decimals());
 
+        let tx_hash_hex = hex::encode(&tx_hash);
+        for rseed in rseeds {
+            info!("saving tx notes rseed for {}!", tx_hash_hex);
+            self.z_fields
+                .locked_notes_db
+                .insert_note(tx.txid().to_string(), rseed)
```

Comment:
Here we store locked notes but the transaction itself is not yet send (unlike in the send_outputs fn). But the user may refuse to send the transaction and we would have notes locked forever.
I think it is better to store notes in the send_raw_tx function, I think we could make a unified function which could decrypt the transaction and store the change output notes.

However, this approach with decrypting tx also changes how to lock spent notes. I guess, we won't be able to get spent notes from raw tx, so we have to get their nullifiers and use them to find notes in the db and mark them 'spent'.

Reply:
nullifier is not unique per note so it can't be used I guess

Reply:
how about we remove locking notes when withdrawn `z_coin`. https://github.com/KomodoPlatform/komodo-defi-framework/pull/2331/commits/07c912b55c0c2aa2c37c3264850e3a8de431ebb9

Reply:
So basically, if user withdraws funds he/she should know that at this time swaps or other withdrawals may fail.
(Maybe good if GUI warns about it, like 'Please wait until this transaction is confirmed before next operations')

Reply:
afaik this applies to all other coins too... if you withdraw during a swap, that swap can fail